### PR TITLE
stop incorrectly adding results to the TestEvent/TestOrder objects

### PR DESCRIPTION
**Pull request by zdeveloper**
# BACKEND PULL REQUEST

## Related Issue
- Fixing bug where the loinc code is missing due to an incorrect java object comparison
- Fixing bug where its adding results from The `TestOrder` into the `TestEvent` object without it actually belonging to the `TestEvent`
https://github.com/CDCgov/prime-simplereport/blob/ff2a9c100384ec84280ce4f03e2fd60edd29a17a/backend/src/main/java/gov/cdc/usds/simplereport/service/TestOrderService.java#L315-L324

Notice how the `TestOrder` Result from line 315 is being fed into `TestEvent` Constructor 

This was leading the TestEvent object to have double the results, and it only has half of that results in the database. 

## Changes Proposed

- remove `TestEvent` Constructors that accepted a Result list
  - you can't construct a `Result` for a `TestEvent` without a `TestEvent` being constructed (aka has an `internal id`). so adding it to the TestEvent constructor is very wrong
- Pulled results creation operations to a `ResultService`, hopefully this reduce complexity going forward.
- Always create separate results for `TestOrder` and `TestEvent` in all tests, unless the ones tests compatibility.

## Testing

- How should reviewers verify this PR? dev6

### CODE REVIEW

**Comment by mehansen** [comment]
partial review - looks good so far! thanks for the explanations as to why it's not working

File: backend/src/test/java/gov/cdc/usds/simplereport/db/repository/ResultRepositoryTest.java [hunk 1]
```diff
@@ -115,6 +115,6 @@ void findAllByDiseaseSuccessful() {
         PATIENT, FACILITY, TestResult.POSITIVE, TestResult.NEGATIVE, TestResult.NEGATIVE, false);
 
     List<Result> results = _repo.findAllByDisease(diseaseService.covid());
-    assertEquals(2, results.size());
+    assertEquals(4, results.size());
```
Comment (zdeveloper) [inline backend/src/test/java/gov/cdc/usds/simplereport/db/repository/ResultRepositoryTest.java:118]: 2 covid results for line 112 (TestEvent result and TestOrder result)
2 covid results for line 114 (TestEvent result and TestOrder result)

File: backend/src/test/java/gov/cdc/usds/simplereport/db/repository/TestEventRepositoryTest.java [hunk 2]
```diff
@@ -82,13 +83,13 @@ void testFindByPatient() {
     _resultRepo.save(positiveResult);
     HashSet positiveResults = new HashSet<>();
     positiveResults.add(positiveResult);
-    _repo.save(new TestEvent(order, false, positiveResults));
```
Comment (zdeveloper) [inline backend/src/test/java/gov/cdc/usds/simplereport/db/repository/TestEventRepositoryTest.java:85]: This is misleading, it doesnt save the `positiveResults` results to the db. Only way to persist a Result entity is via the `resultsRepository.save()`

File: backend/src/main/java/gov/cdc/usds/simplereport/api/converter/FhirConverter.java [hunk 3]
```diff
@@ -488,7 +488,7 @@ public static List<Observation> convertToObservation(
             result -> {
               Optional<DeviceTypeDisease> deviceTypeDisease =
                   deviceType.getSupportedDiseaseTestPerformed().stream()
-                      .filter(code -> code.getSupportedDisease() == result.getDisease())
```
Comment (zdeveloper) [inline backend/src/main/java/gov/cdc/usds/simplereport/api/converter/FhirConverter.java:491]: kinda disappointed sonar didn't catch this :(

File: backend/src/main/java/gov/cdc/usds/simplereport/db/model/Result.java [hunk 4]
```diff
@@ -46,10 +46,8 @@ public class Result extends EternalAuditedEntity {
   @Enumerated(EnumType.STRING)
   private TestResult testResult;
 
-  public Result(
-      TestEvent testEvent, TestOrder testOrder, SupportedDisease disease, TestResult testResult) {
+  public Result(TestEvent testEvent, SupportedDisease disease, TestResult testResult) {
     this.testEvent = testEvent;
-    this.testOrder = testOrder;
     this.disease = disease;
     this.resultLOINC = Translators.convertTestResultToLoinc(testResult);
```
Comment (mehansen) [inline backend/src/main/java/gov/cdc/usds/simplereport/db/model/Result.java:52]: non-blocking nit that can go in another PR but I feel like we should rename resultLOINC to resultSNOMED
I found this confusing and we already have enough LOINCs to worry about

**Review by DanielSass — DISMISSED** [review]

File: backend/src/main/java/gov/cdc/usds/simplereport/db/model/Result.java (hunk 4 above)
Comment (zdeveloper) [reply backend/src/main/java/gov/cdc/usds/simplereport/db/model/Result.java:52]: yeah that a good catch, it will need a db migration to truly change it in the database. I think it belongs in a separate ticket

File: backend/src/test/java/gov/cdc/usds/simplereport/db/repository/TestEventRepositoryTest.java [hunk 5]
```diff
@@ -108,18 +109,20 @@ void testLatestTestEventForPerson() {
     TestOrder firstOrder =
         _dataFactory.createCompletedTestOrder(patient, place, TestResult.POSITIVE);
     var firstResults = firstOrder.getResults();
-    var firstEvent = new TestEvent(firstOrder, false, firstResults);
+    var firstEvent = new TestEvent(firstOrder, false);
+    _repo.save(firstEvent);
     firstResults.forEach(result -> result.setTestEvent(firstEvent));
```
Comment (mehansen) [inline backend/src/test/java/gov/cdc/usds/simplereport/db/repository/TestEventRepositoryTest.java:114]: could we use the new ResultService.addResultsToTestEvent here and in other tests in this file?

File: backend/src/test/java/gov/cdc/usds/simplereport/test_util/TestDataFactory.java [hunk 6]
```diff
@@ -428,13 +428,18 @@ public TestEvent createTestEvent(Person p, Facility f) {
...
-    Result result = new Result(o, diseaseService.covid(), r);
-    TestEvent e = testEventRepository.save(new TestEvent(o, false, Set.of(result)));
+    Result orderResult = new Result(o, diseaseService.covid(), r);
+
+    TestEvent e = new TestEvent(o, false);
+    testEventRepository.save(e);
+
+    Result copiedResult = new Result(orderResult);
+    copiedResult.setTestEvent(e);
+    resultRepository.save(copiedResult);
```
Comment (mehansen) [inline backend/src/test/java/gov/cdc/usds/simplereport/test_util/TestDataFactory.java:438]: do we need to save the `orderResult` in the repo and add it to the order as well? otherwise do we need to create it?

File: backend/src/test/java/gov/cdc/usds/simplereport/test_util/TestDataFactory.java [hunk 7]
```diff
@@ -475,13 +485,17 @@ public TestEvent createMultiplexTestEvent(
...
+        List.of(covid, fluA, fluB).stream().map(Result::new).collect(Collectors.toSet());
+    copiedResults.forEach(result -> result.setTestEvent(event));
+    resultRepository.saveAll(copiedResults);
+    event.getResults().addAll(copiedResults);
+
     resultRepository.save(covid);
-    fluA.setTestEvent(event);
     resultRepository.save(fluA);
-    fluB.setTestEvent(event);
     resultRepository.save(fluB);
```
Comment (mehansen) [inline backend/src/test/java/gov/cdc/usds/simplereport/test_util/TestDataFactory.java:499]: we already saved these at the very beginning of this method right?

**Review by mehansen — DISMISSED** [review]
thanks for fixing! a couple comments on tests but nothing blocking

File: backend/src/main/java/gov/cdc/usds/simplereport/db/model/Result.java (hunk 4 above)
Comment (mehansen) [reply backend/src/main/java/gov/cdc/usds/simplereport/db/model/Result.java:52]: the db col name itself is just `result` which is _technically_ not incorrect but doesn't do much to distinguish it from the other column `test_result` so I could definitely see a ticket for a naming cleanup of these columns and their attributes

**Comment by mehansen** [comment]
also: tested on `dev6` and the bundle looks good

File: backend/src/test/java/gov/cdc/usds/simplereport/test_util/TestDataFactory.java (hunk 7 above)
Comment (zdeveloper) [reply backend/src/test/java/gov/cdc/usds/simplereport/test_util/TestDataFactory.java:499]: oops forgot to remove them

File: backend/src/main/java/gov/cdc/usds/simplereport/db/model/Result.java (hunk 4 above)
Comment (zdeveloper) [reply backend/src/main/java/gov/cdc/usds/simplereport/db/model/Result.java]: I included the name of the field here without the db change

File: backend/src/test/java/gov/cdc/usds/simplereport/db/repository/TestEventRepositoryTest.java (hunk 5 above)
Comment (zdeveloper) [reply backend/src/test/java/gov/cdc/usds/simplereport/db/repository/TestEventRepositoryTest.java]: done

**Review by DanielSass — APPROVED** [review]

**Review by mehansen — APPROVED** [review]
🙌 thanks for all the test cleanup!
